File: lib/domain/usecases/repository/update_repository_usecase.dart
import '../../entities/repository.dart';
import '../../repositories/i_repository_repository.dart';

/// 更新仓库用例
class UpdateRepositoryUseCase {
  UpdateRepositoryUseCase(this._repository);

  final IRepositoryRepository _repository;

  Future<Repository> execute({
    required String id,
    String? name,
    String? icon,
    String? color,
    bool? isArchived,
  }) async {
    final existing = await _repository.getById(id);
    if (existing == null) {
      throw ArgumentError('仓库不存在: $id');
    }

    final updated = existing.copyWith(
      name: name ?? existing.name,
      icon: icon ?? existing.icon,
      color: color ?? existing.color,
      isArchived: isArchived ?? existing.isArchived,
    );

    return await _repository.update(updated);
  }
}

File: web/src/application/usecases/repository/update-repository-usecase.ts
import { Repository } from '../../../domain/entities/repository';
import { IRepositoryRepository } from '../../../domain/repositories/i-repository-repository';
import { validateNotEmpty } from '../../../core/utils/validators';

export interface UpdateRepositoryInput {
  id: string;
  name?: string;
  description?: string | null;
  defaultBranchId?: string | null;
  icon?: string;
  color?: string;
  isArchived?: boolean;
}

export class UpdateRepositoryUseCase {
  constructor(private repositoryRepo: IRepositoryRepository) {}

  async execute(input: UpdateRepositoryInput): Promise<Repository> {
    validateNotEmpty(input.id, '仓库 ID');
    const repo = await this.repositoryRepo.getById(input.id);
    if (!repo) {
      throw new Error('仓库不存在');
    }
    if (input.name !== undefined) {
      validateNotEmpty(input.name, '仓库名称');
    }
    const updated = repo.copyWith(input);
    return this.repositoryRepo.update(updated);
  }
}
